walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,117 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    HashSet<String> attendees = new HashSet<String>(request.getAttendees());
+    HashSet<String> optionalAttendees = new HashSet<String>(request.getOptionalAttendees());
+    ArrayList<TimeRange> badTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalBad = new ArrayList<TimeRange>();
+    Collection<TimeRange> range = new ArrayList<TimeRange>();
+    Collection<TimeRange> OptRange = new ArrayList<TimeRange>();
+    long meetingDuration = request.getDuration();
+    int duration = (int) meetingDuration;
+
+    //Base Cases
+    if(request.getAttendees().size() < 1){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if(meetingDuration > 1440){  // 1440 is the timeRange whole day in minutes
```

Comment:
Suggestion: could make 1440 into a private static final variable at the top of your class. I think it's often better to make magic numbers into variables, because it's really easy to change them at the top if you ever needed to. For example, public static final END_OF_DAY = 1440 (or something like that) would make it really clear what the 1440 means.

Reply:
I would have been a good idea to do this so I would not have to use a comment to explain the meaning. Thank you for the overall feedback I found it really helpful!